### PR TITLE
add context in the function

### PR DESCRIPTION
**Pull request by vitkarpov**

### CODE REVIEW

**Comment by vitkarpov** [comment]
@sohho888 nice! 👏

File: debounce.js [hunk 1]
```diff
@@ -2,10 +2,11 @@ function debounce(fn, delay) {
   let timeID = null;
 
   return function (...args) {
+    let context = this;
```
Comment (vitkarpov) [inline debounce.js]: might be `const` instead of `let`. doesn't seem it can be changed

**Comment by vitkarpov** [comment]
@sohho888 please remove "submodule 'tasks'". Shouldn't be in this PR, should it? Then merge

File: debounce.js (hunk 1 above)
Comment (sohho888) [reply debounce.js]: if we define a variable as const, we will not be able to redefine it in the future

Comment (vitkarpov) [reply debounce.js]: where it can be redefined?

Comment (sohho888) [reply debounce.js]: in line 7 we will not be able to override the value. On line 7 we assign the value returned by setTimeout

Comment (vitkarpov) [reply debounce.js]: probably there's a confusion here. I'm talking about `context` variable not `timeID` (which should be redefined indeed).

**Pull request merged**